bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BRecordType.java

```diff
@@ -49,6 +50,7 @@
     public int typeFlags;
     private final boolean readonly;
     private IntersectionType immutableType;
+    public TypedescValue typedescValue;
```

Comment:
So far in our impl, `TypedescValue` knows about `BType` but `BType` doesn't have any reference to Typedesc. Since TypedescValue already has a reference to BType it can create a circular reference, technically it's ok, but it would be nice if we can avoid it.

Better if we can use TypedescValue directly instead of BType when creating filler values.